### PR TITLE
Replaced HttpError with HttpException

### PR DESCRIPTION
**Pull request by PaulWoitaschek**
As of Retrofit 2.2.0, `HttpException` was moved from the adapters to the base package. So it is no longer necessary to use a custom `HttpError`.

### CODE REVIEW

**Comment by gildor** [comment]
Thank you for your contribution.
I would to do the same changes, thank you for help!

File: build.gradle [hunk 1]
```diff
@@ -24,10 +24,15 @@ apply plugin: 'kotlin'
 targetCompatibility = '1.6'
 sourceCompatibility = '1.6'
 
+sourceSets {
```
Comment (gildor) [inline build.gradle]: But kotlin jvm plugin automatically adds sourceSets `src/main/kotlin` and `src/main/test`, it's not necessary to configure them explicitly.

Comment (gildor) [inline build.gradle]: Did you have some issues in your environment with sourceSets?

Comment (PaulWoitaschek) [inline build.gradle]: Intellij didn't compile without.

Comment (JakeWharton) [inline build.gradle]: FWIW IntelliJ worked fine for me without this...

**Comment by tasomaniac** [comment]
I was going to open an issue :) then I saw there are 3 open PRs :+1:

File: build.gradle (hunk 1 above)
Comment (gildor) [inline build.gradle]: Anyway, not a big deal.

**Pull request merged**